### ISSUE TITLE
docs: Add configs from plugin usage for the new config system.

### ISSUE THREAD

**Issue by jjangga0214** [eslint/eslint#16310]
### What problem do you want to solve?

In the new config system, the specification of plugin is different.

For example, a property `configs` does not have any effect.

How plugin works in the new config system should be described in the docs.

### What do you think is the correct solution?

A new page should be introduced, like [this page](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new) is introduced.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by bpmutter** [comment]
> The only difference is in how you use configs that are bundled with the plugin. 

@nzakas could you provide a bit more info on how to use configs that are bundled with the plugin in the new (flat) config? 

then, i'll add the update to the docs

**Comment by nzakas** [comment]
In the old config system (eslintrc), you could define configs inside of plugins like this:

https://eslint.org/docs/latest/developer-guide/working-with-plugins

in the new config system (flat config), there is no longer an “extends” option. Instead, you can just reference the configs property directly and insert the object into the array. So include a plugin in your config file as described here:

 https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#using-plugins-in-your-configuration
 
And then you can use `jsdoc.configs.recommended` as one member of the array that is exported from eslint.config.JS. 

(On the phone right now, so can’t type out a full example. Let me know if that’s not clear enough.)

**Comment by bpmutter** [comment]
if you could provide a full example, that'd be helpful. i'm still not fully grasping the difference here.

**Comment by nzakas** [comment]
It would look like this:

```js
import jsdoc from "eslint-plugin-jsdoc";

export default [
    // replacement for extends -- just drop the object right into the array
    jsdoc.configs.recommended,
    {
        files: ["**/*.js"],
        plugins: {
            jsdoc: jsdoc
        }
        rules: {
            "jsdoc/require-description": "error",
            "jsdoc/check-values": "error"
        }  
    }
];
```
